Structure_Engine/Query/MaterialComposition.cs

```diff
@@ -402,6 +402,69 @@ public static MaterialComposition MaterialComposition(this PileFoundation pileFo
             return Matter.Compute.AggregateMaterialComposition(elements);
         }
 
+        /***************************************************/
+
+        [Description("Returns a Stem's homogeneous MaterialComposition.")]
+        [Input("stem", "The Stem to get material from.")]
+        [Output("materialComposition", "The kind of matter the Stem is composed of.")]
+        public static MaterialComposition MaterialComposition(this Stem stem)
+        {
+            if (stem.IsNull() || stem.Material.IsNull())
+                return null;
+
+            ReinforcementDensity reinfDensity = stem.FindFragment<ReinforcementDensity>();
+
+            return MaterialComposition(stem.Material, reinfDensity);
+        }
+
+        /***************************************************/
+
+        [Description("Returns a RetainingWall's homogeneous MaterialComposition based on the Stem and Footing.")]
+        [Input("retainingWall", "The RetainingWall to get material from.")]
+        [Output("materialComposition", "The kind of matter the RetainingWall is composed of.")]
+        public static MaterialComposition MaterialComposition(this RetainingWall retainingWall)
+        {
+
+            if (retainingWall.IsNull() && retainingWall.Stem.IsNull() && retainingWall.Footing.IsNull())
+                return null;
+
+
+            List<IElementM> elements = new List<IElementM>
+            {
+                retainingWall.Stem,
+                retainingWall.Footing
+            };
+
+            //Case for when ReinforcementDensity is attached to both hihgher and lower level objects.
+            if ((!retainingWall.Stem.FindFragment<ReinforcementDensity>().IsNull() || !retainingWall.Footing.FindFragment<ReinforcementDensity>().IsNull()) && !retainingWall.FindFragment<ReinforcementDensity>().IsNull())
```

Comment:
```suggestion
            if ((retainingWall.Stem.FindFragment<ReinforcementDensity>() != null || retainingWall.Footing.FindFragment<ReinforcementDensity>() != null) && retainingWall.FindFragment<ReinforcementDensity>() != null)
```
When you call `IsNull` this is why you are getting the error message despite the method working because it raises an auto warning. Here I would just check `!= null`, it's a bit easier to read that `!(object == null)`.

Reply:
Cool. Thank you!